# HG changeset patch
# User David Orban <mozilla@floruit.be>
# Date 1765498862 -3600
# Parent  cc012843a7d52c95192f4f4af974a91fe4535c25
Bug 1901678 - Use event description as email body. Introduces pref calendar.item.eventDescriptionAsEmailBody.

Based on https://phabricator.services.mozilla.com/D231007 without the UI parts.
Text content slightly adjusted to maintain original text.

diff --git a/calendar/base/calendar.js b/calendar/base/calendar.js
--- a/calendar/base/calendar.js
+++ b/calendar/base/calendar.js
@@ -165,8 +165,11 @@ pref("calendar.baseview.loglevel", "Warn
 
 // Enables the prompt when deleting from the item views or trees.
 pref("calendar.item.promptDelete", true);
 
+// Uses the event description as the email body.
+pref("calendar.item.eventDescriptionAsEmailBody", false);
+
 // Enables the new extract service.
 pref("calendar.extract.service.enabled", false);
 
 // Number of days to display in the invite attendees interface.
diff --git a/calendar/itip/CalItipEmailTransport.sys.mjs b/calendar/itip/CalItipEmailTransport.sys.mjs
--- a/calendar/itip/CalItipEmailTransport.sys.mjs
+++ b/calendar/itip/CalItipEmailTransport.sys.mjs
@@ -312,9 +312,28 @@ export class CalItipEmailTransport {
         "--Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)\r\n" +
         "Content-type: text/plain; charset=UTF-8\r\n" +
         "Content-transfer-encoding: 8BIT\r\n" +
         "\r\n" +
-        cal.invitation.encodeUTF8(aBody) +
+        cal.invitation.encodeUTF8(aBody);
+      if (
+        Services.prefs.getBoolPref("calendar.item.eventDescriptionAsEmailBody", false) &&
+        itemList.length == 1
+      ) {
+        // In the text/plain part, add the textual description.
+        mailText +=
+          "\r\n\r\n" +
+          cal.invitation.encodeUTF8(itemList[0].descriptionText) +
+          // Then add a text/html part.
+          "\r\n\r\n\r\n" +
+          "--Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)\r\n" +
+          "Content-type: text/html; charset=UTF-8\r\n" +
+          "Content-transfer-encoding: 8BIT\r\n" +
+          "\r\n" +
+          cal.invitation.encodeUTF8(aBody).replace(/<(.*)>/, "<a href='mailto:$1'>&lt;$1&gt;</a>") +
+          "\r\n<br><br>\r\n" +
+          cal.invitation.encodeUTF8(itemList[0].descriptionHTML);
+      }
+      mailText +=
         "\r\n\r\n\r\n" +
         "--Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)\r\n" +
         "Content-type: text/calendar; method=" +
         aItipItem.responseMethod +
diff --git a/mail/components/MailGlue.sys.mjs b/mail/components/MailGlue.sys.mjs
--- a/mail/components/MailGlue.sys.mjs
+++ b/mail/components/MailGlue.sys.mjs
@@ -1534,8 +1534,9 @@ function reportPreferences() {
 
     // Calendar general
     "calendar.item.editInTab",
     "calendar.item.promptDelete",
+    "calendar.item.eventDescriptionAsEmailBody",
     "calendar.timezone.useSystemTimezone",
 
     // Alarms
     "calendar.alarms.playsound",
